app/auth/profile/deleteprofile.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, Fragment } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { useRouter } from "expo-router";
+import { z } from "zod";
+import { View, StyleSheet, Text, TouchableOpacity, Modal, Pressable } from "react-native";
+import FormContainer from "../../../components/Forms/FormContainer";
+import FormHeader from "../../../components/Forms/FormHeader";
+import FormField from "../../../components/Forms/TextInput";
+import SubmitButton from "../../../components/Forms/SubmitButton";
+import useProfile from "../../../hooks/useProfile";
+import { useAuthentication } from "../../../providers/AuthenticationProvider";
+import { useToast } from "../../../providers/ToastProvider";
+import { colors, ScaleSize, ScaleSizeH, ScaleSizeW } from "../../../utils/SharedStyles";
+
+const schema = z
+  .object({
+    currentPassword: z.string().trim().min(8, "Indtast nuværende adgangskode"),
+    confirmPassword: z.string().trim(),
+  })
+  .superRefine((data, ctx) => {
+    if (data.currentPassword !== data.confirmPassword) {
+      ctx.addIssue({
+        code: "custom",
+        path: ["confirmPassword"],
+        message: "Adgangskode stemmer ikke overens",
+      });
+    }
+  });
+
+type FormData = z.infer<typeof schema>;
+
+const DeleteProfileScreen: React.FC = () => {
+  const { logout } = useAuthentication();
+  const { deleteUser } = useProfile();
+  const { addToast } = useToast();
+  const router = useRouter();
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isSubmitting, isValid },
+  } = useForm<FormData>({
+    resolver: zodResolver(schema),
+    mode: "onChange",
+  });
+
+  const onSubmit = () => {
+    {
+      setModalVisible(true);
+    }
+  };
+
+  return (
+    <Fragment>
+      <ConfirmationModal
+        modalVisible={modalVisible}
+        setModalVisible={setModalVisible}
+        addToast={addToast}
+        deleteUser={deleteUser}
+        logout={logout}
+      />
+      <FormContainer style={{ padding: 30 }}>
+        <FormHeader title="Slet profil" />
+        <FormField
+          control={control}
+          name="currentPassword"
+          placeholder="Indtast nuværende adgangskode"
+          secureText={true}
+        />
+        <FormField
+          control={control}
+          name="confirmPassword"
+          placeholder="Bekræft adgangskode"
+          secureText={true}
+        />
```

Comment:
I had to try this and it doesn't even check it against the backend. I could write two invalid passwords and still delete my account 🤣

Reply:
Det kan godt være min kode ikke virker perfekt og at der er nogle bugs som jeg har overset men det giver dig ikke ret til at grine af min kode Andreas

Reply:
Der er lige nu ikke et endpoint som tillader at man kan tjekke passwordet